src/org/thoughtcrime/securesms/WebxdcActivity.java

```diff
@@ -149,18 +165,24 @@ public void handleEvent(@NonNull DcEvent event) {
       Log.i(TAG, "handleEvent");
       webView.loadUrl("javascript:window.__webxdcUpdate();");
     } else if ((eventId == DcContext.DC_EVENT_MSGS_CHANGED && event.getData2Int() == dcAppMsg.getId())) {
-      updateTitle();
+      updateTitleAndMenu();
     }
   }
 
-  private void updateTitle() {
+  private void updateTitleAndMenu() {
     Util.runOnAnyBackgroundThread(() -> {
       final JSONObject info = this.dcAppMsg.getWebxdcInfo();
       final String docName = JsonUtils.optString(info, "document");
       final String xdcName = JsonUtils.optString(info, "name");
       final String chatName =  WebxdcActivity.this.dcContext.getChat(WebxdcActivity.this.dcAppMsg.getChatId()).getName();
+      final boolean currHasSourceCodeUrl = !JsonUtils.optString(info, "source_code_url").isEmpty();
```

Comment:
why not instead of saving a boolean, just save the URL and then avoid loading it each time in the menu? anyway it is an static property of the webxdc metadata

Reply:
it is loaded only on tapping, but yes, probably makes sense.

Reply:
i changed that, thanks for the tip!